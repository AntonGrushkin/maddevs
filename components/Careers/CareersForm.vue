<template>
  <section id="careers" class="careers">
    <div class="container">
      <div class="careers__wrapper">
        <img
          class="careers__background-logo"
          src="@/assets/img/Careers/svg/careers_logo.svg"
          alt="Careers Background Image"
        />
        <form class="careers__form">
          <label class="careers__form-name-label form-text"
            >Hello, my name is</label
          >
          <input
            class="careers__form-name-input form-text"
            type="text"
            placeholder="|John Smith"
          />
          <h4 class="careers__form-description form-text">
            I want to work for you as a
            <input
              class="careers__form-position-input form-text"
              type="text"
              placeholder="desired position."
            />
          </h4>
          <h4 class="careers__form-description form-text">
            You can also consider me for your other
          </h4>
          <ul class="careers__position-list">
            <li class="careers__position-item">
              <input
                class="careers__position-input"
                name="position"
                id="senior"
                type="radio"
              />
              <label class="careers__position-label form-text" for="senior"
                >Senior,</label
              >
              <div class="careers__position-check">
                <div class="careers__position-check-inside"></div>
              </div>
            </li>
            <li class="careers__position-item">
              <input
                class="careers__position-input"
                name="position"
                id="middle"
                type="radio"
              />
              <label class="careers__position-label form-text" for="middle"
                >Middle,</label
              >
              <div class="careers__position-check">
                <div class="careers__position-check-inside"></div>
              </div>
            </li>
            <li class="careers__position-item">
              <input
                class="careers__position-input"
                name="position"
                id="junior"
                type="radio"
              />
              <label class="careers__position-label form-text" for="junior"
                >Junior,</label
              >
              <div class="careers__position-check">
                <div class="careers__position-check-inside"></div>
              </div>
            </li>
            <li class="careers__position-item">
              <input
                class="careers__position-input"
                name="position"
                id="intern"
                type="radio"
              />
              <label class="careers__position-label form-text" for="intern"
                >Intern</label
              >
              <div class="careers__position-check">
                <div class="careers__position-check-inside"></div>
              </div>
            </li>
          </ul>
          <h4 class="careers__form-description form-text email-title">
            Please reply to
            <input
              class="careers__form-email-input form-text"
              type="text"
              placeholder="your@mail.com"
            />
          </h4>
          <h4 class="careers__form-description form-text">
            To get more information on my skills, please
          </h4>
          <ul class="careers__form-list form-text">
            <li class="careers__form-list-item">
              – check out my
              <a href="#" class="careers__linkedin-link">LinkedIn profile</a> OR
            </li>
            <li class="careers__form-list-item">
              – <a href="#" class="careers__cv">find attached my CV.</a>
            </li>
          </ul>
        </form>
      </div>
    </div>
  </section>
</template>

<script>
export default {
  name: 'CareersForm',
  data() {
    return {};
  }
};
</script>

<style lang="scss" scoped>
@import '../../assets/styles/vars';

.careers {
  .container {
    position: relative;
  }

  &__wrapper {
    border-radius: $form-border-radius;
    background: $bgcolor--grey-light;
    padding: 63px 102px 102px 92px;
  }

  &__background-logo {
    position: absolute;
    bottom: 0;
    right: 0;
    height: auto;
  }

  &__form-name-label {
  }

  .email-title {
    margin: 80px 0;
  }

  &__form-name-input,
  &__form-email-input,
  &__form-position-input {
    background-color: transparent;
    border: 0px solid;
    height: 65px;
    width: 310px;
    color: $text-color--grey;
  }

  &__position-list {
    position: relative;
    display: flex;
    flex-wrap: wrap;
    list-style: none;
    margin-top: 15px;
    padding: 0;
  }

  &__position-item {
    display: flex;
    align-items: center;
    position: relative;
    margin-right: 18px;
  }

  &__position-check {
    box-sizing: border-box;
    position: absolute;
    border: 2px solid $text-color--grey;
    border-radius: 100%;
    height: 26px;
    width: 26px;
    z-index: 5;
    transition: border 0.15s linear;
    -webkit-transition: border 0.15s linear;
  }

  &__position-input:checked ~ &__position-check {
    border-color: $text-color--red;
    border: 0;
  }

  &__position-input:checked ~ &__position-label {
    color: $text-color--red;
  }

  &__position-input:hover &__position-check {
    border: 2px solid $text-color--red;
  }

  &__position-check::before {
    display: block;
    position: absolute;
    content: '';
    border-radius: 100%;
    height: 26px;
    width: 26px;
    margin: auto;
    transition: background 0.15s linear;
    -webkit-transition: background 0.15s linear;
  }

  &__position-input {
    position: absolute;
    visibility: hidden;
  }

  &__position-label {
    position: relative;
    color: $text-color--grey;
    margin: 0 0 0 35px;
    z-index: 9;
    transition: all 0.15s linear;
  }

  &__position-input:checked ~ &__position-check::before {
    background: $text-color--red;
  }

  &__form-description {
    font-family: 'Hoves-Regular';
    color: $text-color--black;
    font-weight: 500;
  }

  &__linkedin-link,
  &__cv {
    color: $text-color--grey;
  }

  &__linkedin-link {
    text-decoration: none;
  }
}

@media only screen and (max-width: 1024px) {
  .careers {
    &__background-logo {
      height: 250px;
      bottom: 10px;
    }
  }
}

@media only screen and (max-width: 870px) {
  .careers {
    &__background-logo {
      height: 220px;
      bottom: 20px;
      right: -70px;
    }
  }
}

@media only screen and (max-width: 670px) {
  .careers {
    &__background-logo {
      height: 180px;
      bottom: 28px;
    }
  }
}

@media only screen and (max-width: 455px) {
  .careers {
    &__background-logo {
      height: 108px;
      bottom: 42px;
    }
  }
}
</style>
