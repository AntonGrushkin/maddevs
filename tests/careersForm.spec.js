import { shallowMount } from '@vue/test-utils';
import CareersForm from '@/components/Careers/CareersForm';

describe('CareersForm component', () => {
  let wrapper;
  const data = {
    isEmailSent: false,
    fullName: null,
    positionValue: null,
    positionTitle: null,
    email: null,
    emailTo: 'hr@maddevs.io',
    selectedFile: null,
    linkedinProfile: null,
    onSumbit: false,
    form: '',
    radioData: [
      { type: 'senior', label: 'Senior,' },
      { type: 'middle', label: 'Middle,' },
      { type: 'junior', label: 'Junior,' },
      { type: 'intern', label: 'Intern' }
    ]
  };

  beforeEach(() => {
    wrapper = shallowMount(CareersForm, {
      stubs: ['ValidationProvider', 'ValidationObserver'],
      mocks: {
        $refs: {
          nameInput: {
            focus: jest.fn()
          }
        },
        errors: false,
        onSumbit: false
      }
    });
    wrapper.vm.$refs = { 
      form: {
        reset: jest.fn()
      },
      fileInput: {
        reset: jest.fn()
      },
      radioButtons: {
        reset: jest.fn()
      }
    };
    wrapper.vm.$nextTick();
  });

  test('is Vue instance', () => {
    expect(wrapper.exists()).toBeTruthy();
  });

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('should check existence of data', () => {
    expect(wrapper.vm.$data).toStrictEqual(data);
  });

  test('onFileChanged method should set correct position', () => {
    const newPositionValue = null;
    CareersForm.methods.onFileChanged(newPositionValue);
    expect(wrapper.vm.$data.positionValue).toStrictEqual(newPositionValue);
  });

  test('sendData should add new object in $data.form', () => {
    const file = new Blob(['testing'], { type: 'application/pdf' });
    const form = {
      'templateId': 305491,
      'variables': {
        'email': '',
        'emailTo': 'hr@maddevs.io',
        'fullName': '',
        'subject': `Job Candidate Application for ${data.positionTitle}`,
        'positionValue': null,
        'positionTitle': '',
        'linkedinProfile': '',
        'attachment': {
          'base64': '',
          'name': ''
        }
      }
    };
    expect(wrapper.vm.$data.form).toEqual('');
    wrapper.vm.toBase64 = file => new Promise((res, rej) => res('string'));
    wrapper.vm.sendData();
    expect(wrapper.vm.$data.form).toEqual(form);
  });

  test('should rest values in data instances', () => {
    // Set mock data for data instances
    wrapper.vm.$data.fullName = 'Name';
    wrapper.vm.$data.email = 'email@mail.com';
    wrapper.vm.$data.form = {
      value1: 'value1',
      value2: 'value2'
    };

    wrapper.vm.resetForm();
    expect(
      wrapper.vm.$data.fullName &&
      wrapper.vm.$data.email &&
      wrapper.vm.$data.form
    ).toEqual(null);
  });

  test('toBase64 should return promise', () => {
    const file = new Blob(['testing'], { type: 'application/pdf' });
    const promise = new Promise((res, rej) => res());
    expect(wrapper.vm.toBase64(file)).toEqual(promise);
  });
});
