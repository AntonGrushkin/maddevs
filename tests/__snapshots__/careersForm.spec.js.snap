// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CareersForm component renders correctly 1`] = `
<section
  class="careers"
  id="careers"
>
  <div
    class="container"
  >
    <div
      class="careers__wrapper"
    >
      <img
        alt="Careers Background Image"
        class="careers__background-logo"
        src="@/assets/img/Careers/svg/careers_logo.svg"
      />
       
      <validationobserver-stub />
    </div>
  </div>
   
  <successmodal-stub
    id="career-modal"
  />
</section>
`;
